File: app/src/main/java/com/revosleap/bxplayer/AppUtils/Utils/GetAudio.java
package com.revosleap.bxplayer.AppUtils.Utils;

import android.content.Context;
import android.database.Cursor;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.media.MediaMetadataRetriever;
import android.net.Uri;
import android.provider.MediaStore;
import android.util.Log;
import android.widget.Toast;

import com.revosleap.bxplayer.AppUtils.Models.AudioModel;
import com.revosleap.bxplayer.R;

import java.io.ByteArrayInputStream;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;

public class GetAudio {
    public List<AudioModel> geAllAudio(Context context){

        List<AudioModel> temAudioList= new ArrayList<>();
        Uri uri= MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
        String order= MediaStore.Audio.Media.TITLE;

        String[] projection={MediaStore.Audio.AudioColumns.DATA,MediaStore.Audio.AudioColumns.ALBUM,
        MediaStore.Audio.ArtistColumns.ARTIST,MediaStore.Audio.AudioColumns.TITLE,MediaStore.Audio.AudioColumns.DURATION};
        Cursor cursor= context.getContentResolver().query(uri,projection,null,
                null,order);
        if (cursor!=null){
           Log.v("Count","Songs "+cursor.getCount());
            while (cursor.moveToNext()){

                AudioModel model= new AudioModel();
                String path= cursor.getString(0);
                String album= cursor.getString(1);
                String artist= cursor.getString(2);
                String name= cursor.getString(3);
                String duration=cursor.getString(4);

                model.setTitle(name);
                model.setAlbum(album);
                model.setArtist(artist);
                model.setPath(path);
                model.setDuration(Integer.parseInt(duration));


                temAudioList.add(model);
            }
            cursor.close();
        }
        else Toast.makeText(context, "Null", Toast.LENGTH_SHORT).show();
        return temAudioList;

    }
}

File: app/src/main/java/com/revosleap/bxplayer/utils/utils/GetAudio.kt
package com.revosleap.bxplayer.utils.utils

import android.content.Context
import android.provider.MediaStore
import android.util.Log
import android.widget.Toast
import com.revosleap.bxplayer.utils.models.AudioModel
import java.util.*

class GetAudio {
    fun geAllAudio(context: Context): List<AudioModel> {

        val temAudioList = ArrayList<AudioModel>()
        val uri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI
        val order = MediaStore.Audio.Media.TITLE

        val projection = arrayOf(MediaStore.Audio.AudioColumns.DATA, MediaStore.Audio.AudioColumns.ALBUM, MediaStore.Audio.ArtistColumns.ARTIST, MediaStore.Audio.AudioColumns.TITLE, MediaStore.Audio.AudioColumns.DURATION)
        val cursor = context.contentResolver.query(uri, projection, null, null, order)
        if (cursor != null) {
            Log.v("Count", "Songs " + cursor.count)
            while (cursor.moveToNext()) {

                val model = AudioModel()
                val path = cursor.getString(0)
                val album = cursor.getString(1)
                val artist = cursor.getString(2)
                val name = cursor.getString(3)
                val duration = cursor.getString(4)

                try {
                    model.title = name

                    model.album = album
                    model.artist = artist
                    model.path = path
                    model.duration = Integer.parseInt(duration)
                } catch (e: Exception) {
                    e.printStackTrace()
                }


                temAudioList.add(model)
            }
            cursor.close()
        } else
            Toast.makeText(context, "Null", Toast.LENGTH_SHORT).show()
        return temAudioList

    }
}
